File: sanket-dalvi/Day7-8/mergeIntervals.py
from typing import List


class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort(key=lambda x: (x[0],-x[1]))
        res = [intervals[0]]
        for i in range(1, len(intervals)):
            if res[-1][0] <= intervals[i][0] <= res[-1][1]:
                res[-1][1] = max(intervals[i][1], res[-1][1])
            else:
                res.append(intervals[i])
        return res

# Test
print(Solution().merge([[1,3],[2,6],[8,10],[15,18]]))

File: Adhiti_Shaili/Day 7,8/Merge Intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        vector<vector<int>> ans;
        sort(intervals.begin(),intervals.end());
        int s=intervals[0][0];
        int e=intervals[0][1];
        
        for(int i=1;i<intervals.size();i++)
        {
            if(intervals[i][0]<=e)
            {
                e=max(e,intervals[i][1]);
            }
            else
            {
                ans.push_back({s,e});
                s=intervals[i][0];
                e=intervals[i][1];
            }
        }
        ans.push_back({s,e});
        return ans;
    }
};